ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -779,29 +782,59 @@ private AdminResponse handleBlobStoreControlRequest(DataInputStream requestStrea
     ServerErrorCode error;
     BlobStoreControlAdminRequest blobStoreControlAdminRequest =
         BlobStoreControlAdminRequest.readFrom(requestStream, adminRequest);
-    if (blobStoreControlAdminRequest.getNumReplicasCaughtUpPerPartition() > 0) {
-      PartitionId partitionId = blobStoreControlAdminRequest.getPartitionId();
-      if (partitionId != null) {
-        error = validateRequest(partitionId, RequestOrResponseType.AdminRequest);
+    PartitionId partitionId = blobStoreControlAdminRequest.getPartitionId();
+    if (partitionId != null) {
+      if (blobStoreControlAdminRequest.shouldEnable()) {
+        // start BlobStore properly
+        error = validateRequest(partitionId, RequestOrResponseType.AdminRequest, true);
         if (error.equals(ServerErrorCode.No_Error)) {
-          if (blobStoreControlAdminRequest.shouldEnable()) {
-            // TODO: start BlobStore properly
-            error = ServerErrorCode.Temporarily_Disabled;
-          } else {
+          if (storageManager.startBlobStore(partitionId)) {
```

Comment:
As in the case of shutting down, we should make sure that the `ClusterParticipant` is made aware of this, so it can synchronously update the status of the store/replica in Helix.

Reply:
I think that we should reserve all clustermap interaction to another PR.

Reply:
Changes related to clustermap have been mentioned in issue #882 ,  which will be made in a new PR.